### PR TITLE
[2.0.1] high: make fully buildable again

### PR DESCRIPTION
**Pull request by jnpkrn**
Wasn't the case for Fedora Rawhide for two reasons:
- test output neutralizing neglected time-zone-full cases
- overall design neglected variance brought into the game
  when hash tables are used + glib 2.59.0 introduced
  exactly such a variance

Regarding the latter, some fundamental assumptions in
pacemaker may be broken (would be serious with a wide
reach), but for now, we just treat it as that's the actual tests
(due to flawed approach used to test) that is a broken
thing here.

### CODE REVIEW

**Comment by kgaillot** [comment]
All good except d76a261. I don't want to automatically disable the tests; they are legitimately failing, and the user needs to know that. I'd rather a new cts-cli option (e.g. --skip-glib-2.59), so they have to explicitly know what they're doing. Even that should be marked experimental and should be removed when we fix the underlying issue.

If you want to pull out that one commit here for a new PR, I'll merge the rest.

**Comment by jnpkrn** [comment]
On 21/01/19 18:44 +0000, Ken Gaillot wrote:
> All good except d76a261. I don't want to automatically disable the
> tests; they are legitimately failing, and the user needs to know
> that.

They only get disabled, when triggering glib is detected in the
configure time, or when one disables them explicitly.

I don't see a problem with that, especially since downstream can
set its preference explicitly.

> I'd rather a new cts-cli option (e.g. --skip-glib-2.59), so
> they have to explicitly know what they're doing. Even that should be
> marked experimental and should be removed when we fix the underlying
> issue.

It's really questionable what's actually broken here.
Pacemaker internals?  Just the tests?

> If you want to pull out that one commit here for a new PR, I'll
> merge the rest.

I think it may stay as-is since it allows for enough of flexibility,
fixes the immedate hiccup, and can silently be no-op'ized once there's
no need for that to intervene -- just as we had since-ever-futile
--with-pkgname switch until dropped in #1669.

There's always a trade-off to be made, I consider this a best interim
solution.

-- 
Nazdar,
Jan (Poki)

**Comment by kgaillot** [comment]
> They only get disabled when triggering glib is detected at the configure time,

That's what I want to avoid. The tests are applicable to the local platform, so skipping them automatically is problematic. The user should be made aware of the issue and be required to take explicit action.

> It's really questionable what's actually broken here. Pacemaker internals? Just the tests?

Exactly, that's one reason they shouldn't be disabled automatically. I expect that we we will find that it is only the tests, but we haven't confirmed that yet. Even if it does end up being testing only, disabling the tests means the functionality they are meant to verify is not verified, and again the user should know that.

The bottom line is that we need to fix the incompatibility rather than work around it. Any user or distro with a newer gcc can skip the scheduler testing or apply local patches as desired in the meantime.

**Comment by kgaillot** [comment]
Also, the sed changes here are not working on FreeBSD

**Comment by jnpkrn** [comment]
On 24/01/19 07:27 -0800, Ken Gaillot wrote:
> Also, the sed changes here are not working on FreeBSD

Oops, have I been testing with `-t dates` instead?
Anyway, fixed with overapproximating zero-or-one with
zero-or-more.

-- 
Jan (Poki)

**Comment by jnpkrn** [comment]
One has to pass `----disable-hash-affected-tests=try` explicitly to have the
automatism in place.  Hopefully this answers the immediate concern.
We currently have no idea about what the future solution will be.

**Comment by kgaillot** [comment]
Great, thanks

**Comment by jnpkrn** [comment]
I've noticed I must have missed a single test -- thanks for correcting
that.  I figure it might have been caused with the psychologically
prohibitive distraction that rerunning `cts-scheduler` suite imposes
on me (if not anyone else), so better to do me/us a favour and unlock
the potential of this suite fully (on a multi-core machine, anyway):
#1688
